### PR TITLE
Strip spaces from room codes

### LINKED ISSUE #51: Disallow spaces in room codes

### PR DESCRIPTION
**Pull request by jacebrowning**
Fixes #51

### CODE REVIEW

File: kcards/templates/join.html [hunk 1]
```diff
@@ -3,9 +3,13 @@
...
-</form>
+    <p>What is your name?</p>
+    <form method="post">
+        <input type="text" name="name" placeholder="Firstname Lastname"></input>
+        <input type="submit" name="join" value="Join Room"></input>
+    </form>
 {% endblock%}
+
+{% block footer %}
+    <div class="footer">{{ code }}</div>
```
Comment (jacebrowning) [inline kcards/templates/join.html]: We need the room code to appear somewhere on the page for testing. I think ultimately we'll want the room code on the page somewhere for the user.

File: kcards/tests/test_models_room.py [hunk 2]
```diff
@@ -22,18 +22,24 @@ def it_generates_an_id_when_unspecified():
...
             expect(room1) != room2
 
+        def it_allows_custom_codes():
+            room = Room(code='1234')
+
+            expect(room.code) == '1234'
+
+        def it_replaces_spaces_with_dashes():
+            room = Room(code='with spaces  here')
+            room.clean()
```
Comment (jacebrowning) [inline kcards/tests/test_models_room.py]: `.clean()` is called before `.save()` by MongoEngine.

File: kcards/tests/test_models_room.py [hunk 3]
```diff
@@ -22,18 +22,24 @@ def it_generates_an_id_when_unspecified():
...
     def describe_sort():
 
         def it_uses_the_id():
             rooms = [Room(code='1'), Room(code='A'), Room(code='a')]
 
             expect(sorted(rooms)) == rooms
 
-    def describe_code():
-
-        def it_matches_the_id(room):
```
Comment (jacebrowning) [inline kcards/tests/test_models_room.py]: This test wasn't really doing much.

File: kcards/views/api_rooms.py [hunk 4]
```diff
@@ -45,7 +45,7 @@ def cleanup():
 
 @blueprint.route("/<code>")
 def detail(code):
-    room = Room.objects(code=code).first()
+    room = Room.objects(code=Room.clean_code(code)).first()
```
Comment (jacebrowning) [inline kcards/views/api_rooms.py]: I am not sure if this is the best place to do this nor if there are other places we'll need to cleanup incoming codes from the user.

**Comment by jacebrowning** [comment]
@DanLindeman There are still a few things I'm not totally set on here, but I think this is ready for review. To test:

- [ ] Create rooms with and without spaces in them
- [ ] Join those rooms using the code with and without spaces

**Comment by jacebrowning** [comment]
@DanLindeman How did you make a room with only spaces. I see "Room code is required." when trying to create a room with only spaces on http://kcards-pr-74.herokuapp.com.

**Comment by DanLindeman** [comment]
I made a mistake. The check does appear to block me from _creating_ a room with only spaces, Joining 404s....which makes sense because the room doesn't exist yet.

**Comment by jacebrowning** [comment]
I'll add a message when trying to join a room with an empty code...

**Pull request merged**